src/js/utils/html-utils.js

```diff
@@ -107,17 +147,70 @@
                         {
                             HtmlUtils: HtmlUtils
                         },
-                        data
+                        data || {}
                     );
-                    return rawTemplate(augmentedData);
+                    return HTML(rawTemplate(augmentedData));
                 };
             return template;
         };
 
+        /**
+         * A wrapper for $.html that safely escapes the provided HTML.
+         *
+         * If the HTML is provided as an HTML snippet then it is used directly.
+         * If the value is a string then it is assumed to be unescaped and
+         * so it is first escaped before being used.
+         *
+         * @param {element} element The element or elements to be updated.
+         * @param {(string|HtmlSnippet)} html The desired HTML, either as a
+         * plain string or as an HTML snippet.
+         * @returns {JQuery} The JQuery object representing the element or elements.
+         */
+        setHtml = function(element, html) {
+            return $(element).html(ensureHtml(html).toString());
+        };
+
+        /**
+         * A wrapper for $.append that safely escapes the provided HTML.
+         *
+         * If the HTML is provided as an HTML snippet then it is used directly.
+         * If the value is a string then it is assumed to be unescaped and
```

Comment:
See my comment on the other PR: https://github.com/edx/edx-platform/pull/11856/files#r57047568

IMO it is useful for ``setHtml`` and arguable for ``append``. I'd like to keep it for consistency and to enforce safety as you point out.